File: Excel/800-899/895/Challenge 895.R
library(tidyverse)
library(readxl)

path <- "Excel/800-899/895/895 Max Sales.xlsx"
input <- read_excel(path, range = "A2:F51")
test <- read_excel(path, range = "H2:J9")

result = input %>%
  mutate(Amount = Price * Units, Quarter = floor_date(Date, "quarter")) %>%
  group_by(Quarter, SalesRep) %>%
  summarise(Total_Sales = sum(Amount)) %>%
  filter(Total_Sales == max(Total_Sales)) %>%
  ungroup() %>%
  summarise(
    Name = str_c(SalesRep, collapse = ", "),
    Amount = first(Total_Sales),
    .by = Quarter
  ) %>%
  mutate(Quarter = paste0("Q", quarter(Quarter), "-", year(Quarter)))

all.equal(result, test)
# [1] TRUE

File: Excel/800-899/895/Challenge 895.py
import pandas as pd

path = "Excel/800-899/895/895 Max Sales.xlsx"
input = pd.read_excel(path, usecols="A:F", skiprows=1, nrows=50)
test = pd.read_excel(path, usecols="H:J", skiprows=1, nrows=7)

input['Amount'] = input['Price'] * input['Units']
input['Quarter'] = pd.to_datetime(input['Date'])
input['Quarter'] = input['Quarter'].dt.to_period('Q').dt.start_time
input['Amount'] = input['Amount'].astype('int64')

grouped = input.groupby(['Quarter', 'SalesRep'], as_index=False)['Amount'].sum()
grouped = grouped.rename(columns={'Amount': 'Total_Sales'})

max_sales = grouped.groupby('Quarter')['Total_Sales'].transform('max')
filtered = grouped[grouped['Total_Sales'] == max_sales]

result = filtered.groupby('Quarter', as_index=False).agg({
    'SalesRep': lambda x: ', '.join(x),
    'Total_Sales': 'first'
})

result['Quarter'] = result['Quarter'].dt.to_period('Q')
result['Quarter'] = result['Quarter'].apply(lambda x: f"Q{x.quarter}-{x.year}")
result = result.rename(columns={'SalesRep': 'Name', 'Total_Sales': 'Amount'})

print(result.equals(test))
# True